File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
def solution(stones, k):
    answer = 0
    
    def check(stones, minus):
        n = 0
        for s in stones:
            if s - minus <= 0:
                n += 1
            else:
                n = 0
            if n == k: return True
        return False

    left, right = 1, 200_000_000
    while left <= right:
        mid = (left + right) // 2

        if check(stones, mid):
            right = mid - 1
        else:
            left = mid + 1
    return left

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.kt
class Solution {
    fun solution(stones: IntArray, k: Int): Int {
        var (left, right) = Pair(0, 200_000_000)
        
        while (left <= right) {
            val mid = (left + right) / 2
            if (check(stones, mid, k)) {
                right = mid - 1
            } else {
                left = mid + 1
            }
        }
        return left
    }

    private fun check(li: IntArray, minus: Int, k: Int): Boolean {
        var count = 0
        for (x in li) {
            if (x - minus <= 0) {
                count++
            } else {
                count = 0
            }
            if (count == k) return true
        }
        return false
    }
}